examples/quickstart_gcp/quickstart_gcp/io_managers.py

```diff
@@ -46,7 +46,9 @@ def handle_output(self, context: OutputContext, obj: pd.DataFrame):
 
     def load_input(self, context: InputContext):
         # upstream_output.asset_key is the asset key given to the Out that we're loading for
-        table_name = context.upstream_output.asset_key.to_python_identifier()
+        output = context.upstream_output
+        assert output is not None
```

Comment:
We generally avoid asserts in non-test code. This should be able to just use `context.asset_key` though.

Reply:
Unfortunately there is no good alternative to `assert` that I'm aware of for type-narrowing:

```
def some_fn() -> Union[Foo, Bar]:
    ...

x = some_fn()  # x has type Union[Foo, Bar]
assert isinstance(x, Foo)
x  # x now has type Foo
```

`check.inst` does not work here, and there is no way to make it work that I'm aware of (and I've tried quite a bit).